aepp/exportDatasetToCloudStorage.py

```diff
@@ -0,0 +1,179 @@
+            data={
+                "name": f"[CMLE][Week2] Data Landing Zone target connection created by {self.username}",
+                "baseConnectionId": base_connection_id,
+                "params": {
+                    "mode": "Server-to-server",
+                    "compression": compression_type,
+                    "datasetFileType": data_format,
+                    "path": export_path
+                },
+                "connectionSpec": {
+                    "id": connection_spec_id,
+                    "version": "1.0"
+                }
+            }
+        )
+        try:
+            target_connection_id = target_res["id"]
+            print("targetconnectionId " + target_connection_id + " has been created")
+        except KeyError:
+            raise RuntimeError(f"Error when creating target connection: {target_res}")
+
+        if on_schedule:
+            schedule_params = {
+                "interval": 3,
+                "timeUnit": "hour",
+                "startTime": int(time.time())
+            }
+        else:
+            schedule_params = {
+                "interval": 1,
+                "timeUnit": "day",
+                "startTime": int(time.time() + 60*60*24*365)
+            }
+        flow_spec_id = "cd2fc47e-e838-4f38-a581-8fff2f99b63a"
+        flow_obj = {
+            "name": f"[CMLE][Week2] Flow for Featurized Dataset to DLZ created by {self.username}",
+            "flowSpec": {
+                "id": flow_spec_id,
+                "version": "1.0"
+            },
+            "sourceConnectionIds": [
+                source_connection_id
+            ],
+            "targetConnectionIds": [
+                target_connection_id
+            ],
+            "transformations": [],
+            "scheduleParams": schedule_params
+        }
+        flow_res = self.flow_conn.createFlow(
+            obj = flow_obj,
+            flow_spec_id = flow_spec_id
+        )
+        try:
+            dataflow_id = flow_res["id"]
+            print("dataflowId " + dataflow_id + " has been created")
+        except KeyError:
+            raise RuntimeError(f"Error when creating data flow: {flow_res}")
+
+        #save dataflow_id to config
+        self.utils.save_field_in_config("Platform", "dataflow_id", dataflow_id)
+        return dataflow_id
+
+    def createFlowRun(self, dataflow_id, dataset_id, on_schedule):
+        if not on_schedule:
+            res = self.retryOnNotReadyException(dataflow_id = dataflow_id, dataset_id= dataset_id)
+            flowRunId = res["destinations"][0]["datasets"][0]["statusURL"].rsplit('/', 1)[-1]
+            #check if flowRun has finished
+            finished = False
+            while not finished:
+                try:
+                    run = self.flow_conn.getRun(runId = flowRunId)["items"][0]
+                    run_id = run["id"]
+                    run_started_at = run["metrics"]["durationSummary"]["startedAtUTC"]
```

Comment:
Are the metrics entries guaranteed to all be here? I would recommend doing a check and if a given metric field does not exist just ignore it.

Reply:
All metrics entries are expected to show up.